webapp/src/utils/validationUtils.ts

```diff
@@ -0,0 +1,192 @@
+import { t } from "i18next";
+
+////////////////////////////////////////////////////////////////
+// Types
+////////////////////////////////////////////////////////////////
+
+interface ValidationOptions {
+  existingValues?: string[];
+  excludedValues?: string[];
+  isCaseSensitive?: boolean;
+  allowSpecialChars?: boolean;
+  allowedChars?: string;
+  allowSpaces?: boolean;
+  editedValue?: string;
+  min?: number;
+  max?: number;
+}
+
+////////////////////////////////////////////////////////////////
+// Validators
+////////////////////////////////////////////////////////////////
+
+/**
+ * Validates a single string value against specified criteria.
+ *
+ * Validates the input string against a variety of checks including length restrictions,
+ * character validations, and uniqueness against provided arrays of existing and excluded values.
+ *
+ * @param value - The string to validate. Leading and trailing spaces will be trimmed.
+ * @param options - Configuration options for validation.
+ * @param options.existingValues - An array of strings to check against for duplicates. Comparison is case-insensitive by default.
+ * @param options.excludedValues - An array of strings that the value should not match.
+ * @param options.isCaseSensitive - Whether the comparison with `existingValues` and `excludedValues` is case-sensitive. Defaults to false.
+ * @param options.allowSpecialChars - Flags if special characters are permitted in the value.
```

Comment:
`allowSpecialChars` and `allowedChars` can be fusioned:

allowSpecialChars?: boolean | string

If true, all special chars are allowed, if string, only the specified ones.

`allowedChars` name is confused.

Reply:
Don't know about this, 🤔  it's confusing to have one parameter handle 2 data types and use cases in my opinion.

While combining parameters might seem like a simplification, it can actually complicate usage and understanding for those implementing the API. Detailed, specific parameters can help leading to fewer mistakes and more predictable outcomes, separate concerns and have more granular control.

Reply:
`allowedChars` => `allowedSpecialChars` ?

Reply:
It is common to have parameter with mutilple types.

`allowedChars` is not clear because letters and digit are chars too.
And it's dependant of `allowSpecialChars` value.

Reply:
Edit: the `allowedChars` parameter is now renamed `specialChars` for more clarity as you suggested, thanks.

I tried to merge the flag `allowSpecialChars` and `specialChars`, but this change doesn't align with the original design, I don't want the possibility to allow all special characters for obvious raison, this is against the concept of validation, and can cause potential security risks.

So for now we have 2 possible cases:

- 1 Special chars blocked:
```javascript
validateString(v, { allowSpecialChars: false })
```

OR

- 2 Only a specific set of special chars are allowed (by default `"&()_-"` to avoid repeating this common pattern all around):
```javascript
validateString(v) // will block `"&()_-"`
```
 and that can be adjusted as needed, e.g. for an email input:
 
```javascript
validateString(v, { specialChars: "@" })
```

What do you think about that ?